src/models/repo/posts.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+class PostsRepo {
+    /**
+     * ユーザーIDを元に投稿数を取得
+     *
+     * @param string $userId ユーザーID
+     * @return integer 投稿数
+     */
+    public static function getPostsCountByUserId(string $userId): int {
+        $db = PDOFactory::getPDOInstance();
+        $postsDao = new PostsDAO($db);
+        return $postsDao->getPostsCountByUserId($userId);
+    }
```

Comment:
DB操作はRepoで行うようにして良いですが、このように書くとしたら、テストコードは書きにくいと思います。
PDOにもDIを使うと、テストコードを書きやすいし、毎回PDOインスタンスを取得することも避けられます。
```suggestion
    public static function getPostsCountByUserId(string $userId, PDO $pdo = null): int {
       if (is_null($pdo)) $pdo = PDOFactory::getPDOInstance();
        $postsDao = new PostsDAO($db);
        return $postsDao->getPostsCountByUserId($userId);
    }
```